### PR TITLE
Feature/main page sort

### PR DESCRIPTION
**Pull request by denisVorobyovExceed**
Fixed sort bug finally

### CODE REVIEW

File: src/appointment/Appointment/Appointment.js [hunk 1]
```diff
@@ -1,38 +1,79 @@
...
-      'headers': { 'Authorization': authStr }
-    }).then(res => {
-      if (res.status === 200) {
-        setDoctors(res.data);
-      }
-    }).catch(error => {
-      //handle later
+  const getDoctorsName = (id) => {
+    let fullName = '';
+    doctors.forEach(doctor => {
```
Comment (usatovaDasha) [inline src/appointment/Appointment/Appointment.js:42]: change on filter

File: src/appointment/Buttons/Buttons.js [hunk 2]
```diff
@@ -1,16 +1,62 @@
...
 import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+import DeleteDialog from '../DeleteDialog/DeleteDialog';
+import EditDialog from '../EditDialog/EditDialog';
 
-function Buttons() {
+export default function Buttons({ appointment, doctors, getAppointments }) {
+  const [editOpen, setEditOpen] = useState(false);
+  const [deleteOpen, setDeleteOpen] = useState(false);
+
+
```
Comment (usatovaDasha) [inline src/appointment/Buttons/Buttons.js:13]: delete empty line

File: src/appointment/Buttons/Buttons.js [hunk 3]
```diff
@@ -1,16 +1,62 @@
...
+    });
+  }
 
   return (
     <div className='buttons'>
-      <EditIcon />
-      <DeleteOutlineIcon />
+      <EditIcon onClick={() => setEditOpen(true)} />
+      <DeleteOutlineIcon onClick={() => setDeleteOpen(true)} />
+
```
Comment (usatovaDasha) [inline src/appointment/Buttons/Buttons.js:47]: delete empty line

File: src/appointment/DeleteDialog/DeleteDialog.js [hunk 4]
```diff
@@ -1,46 +1,50 @@
...
-        onClose={handleClose}
-        aria-labelledby="alert-dialog-title"
-        aria-describedby="alert-dialog-description"
+        onClose={() => setOpen(false)}
+        aria-labelledby="delete-dialog-title"
+        aria-describedby="delete-dialog-description"
       >
-        <DialogTitle id="alert-dialog-title">{"Use Google's location service?"}</DialogTitle>
+        <DialogTitle id="delete-dialog-title" >
+          {"Удалить прием"}
```
Comment (usatovaDasha) [inline src/appointment/DeleteDialog/DeleteDialog.js:23]: don't need js brackets

File: src/appointment/EditDialog/EditDialog.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  MenuItem,
+  Dialog, 
+  DialogActions, 
+  DialogContent,  
+  DialogTitle 
+} from '@material-ui/core';
+import './EditDialog.scss';
+
+export default function EditDialog({ appointment, doctors, open, saveApp, setOpen }) {
+  const getDoctorsName = (id) => {
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.js:15]: move the function below the state

File: src/appointment/EditDialog/EditDialog.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+  useEffect( () => {
+    if (patientName && doctorName && dateOfAppointment && complaints) {
+      setDisabled(false);
+    } else setDisabled(true);
+  }, [patientName, doctorName, dateOfAppointment, complaints]);
+
+  const getDoctorId = () => {
+    let id = 0;
+    doctors.forEach(doctor => {
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.js:49]: change on filter

File: src/appointment/EditDialog/EditDialog.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+  return (
+    <div>
+      <Dialog
+        open={open}
+        onClose={() => setOpen(false)}
+        aria-labelledby="edit-dialog-title"
+        aria-describedby="edit-dialog-description"
+      >
+        <DialogTitle id="edit-dialog-title">
+          {"Изменить прием"}
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.js:78]: don't need js brackets

File: src/appointment/EditDialog/EditDialog.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+        </DialogTitle>
+        <DialogContent id="edit-dialog-content">
+
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.js:81]: remove empty line

File: src/appointment/EditDialog/EditDialog.scss [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+.MuiInputBase-root {
+  input {
+    font-family: 'Roboto' !important;
+    font-style: normal;
+    font-weight: 300;
+    font-size: 18px; 
+    color: #000000;
+  }
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.scss:44]: add empty line

File: src/appointment/List/List.js [hunk 10]
```diff
@@ -26,21 +20,45 @@ function List({ appointments, doctors }) {
...
-              {(new Date( Date.parse(item.date) )).toLocaleDateString('ru-RU')}
-            </td>
-            <td className='app-cell comp'>{item.complaints}</td>
-            <td className='app-cell'><Buttons /></td>
-          </tr>
-        ))}
+          <SingleRow {...props}/>
         </tbody>
       </table>
+      {props.appointments === [] ? <p>Записи отсутствуют</p> : false}
```
Comment (usatovaDasha) [inline src/appointment/List/List.js:26]: check length

File: src/appointment/List/List.js [hunk 11]
```diff
@@ -26,21 +20,45 @@ function List({ appointments, doctors }) {
...
     </Container>
-  )
+  );
 }
+//move it back later:
+function SingleRow({ appointments, doctors, getAppointments }) {
+  const getDoctorsName = (id) => {
+    let fullName = '';
+    doctors.forEach(doctor => {
```
Comment (usatovaDasha) [inline src/appointment/List/List.js:34]: change on filter

File: src/appointment/NewTab/NewTab.scss [hunk 12]
```diff
@@ -1,3 +1,7 @@
+@media (min-width: 720px) {
```
Comment (usatovaDasha) [inline src/appointment/NewTab/NewTab.scss:1]: ???

File: src/appointment/NewTab/NewTab.scss [hunk 13]
```diff
@@ -30,6 +34,7 @@
 
 .MuiTextField-root {
   width: 99%;
+  max-width: 300px;
```
Comment (usatovaDasha) [inline src/appointment/NewTab/NewTab.scss:37]: px?

File: src/appointment/SortBar/SortBar.scss [hunk 14]
```diff
@@ -0,0 +1,43 @@
+.sort-bar-container.MuiContainer-root {
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: flex-end;
+  padding-top: 39px;
```
Comment (usatovaDasha) [inline src/appointment/SortBar/SortBar.scss:6]: change on 40px

File: src/appointment/SortBar/SortBar.scss [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+#select-sort {
+  max-width: 200px !important;
```
Comment (usatovaDasha) [inline src/appointment/SortBar/SortBar.scss:10]: px? remove  all important

File: src/common/Header/Header.scss [hunk 16]
```diff
@@ -37,4 +37,13 @@
   font-size: 36px;
   line-height: 42px;
   color: #000000;
+}
+
+.button-reg.MuiButton-root {
+  width: 90px !important;
```
Comment (usatovaDasha) [inline src/common/Header/Header.scss:43]: px?

File: src/appointment/NewTab/NewTab.scss (hunk 12 above)
Comment (denisVorobyovExceed) [reply src/appointment/NewTab/NewTab.scss:1]: Oops

File: src/appointment/SortBar/SortBar.scss (hunk 15 above)
Comment (denisVorobyovExceed) [reply src/appointment/SortBar/SortBar.scss:10]: Remove all is not an option, but I'll try to remove all I can

File: src/appointment/EditDialog/EditDialog.js [hunk 17]
```diff
@@ -11,14 +11,9 @@ import {
...
 
-export default function EditDialog({ appointment, doctors, open, saveApp, setOpen }) {
-  const getDoctorsName = (id) => {
-    let fullName = '';
-    doctors.forEach(doctor => {
-      if (doctor._id === id) fullName = doctor.name;
-    });
-    return fullName;
-  }
+export default function EditDialog(
```
Comment (usatovaDasha) [inline src/appointment/EditDialog/EditDialog.js:14]: EditDialog({
appointment,
..
})

**Pull request merged**